generate.go

```diff
@@ -190,6 +191,7 @@ func generateConversion(cfg structConfig, t targetStruct, imports *imports) (gen
 				sourceField.ConvertFuncName(DirFrom),
 				kind.Direct,
 				kind.Convert,
+				kind.Special.ReverseDirection(),
```

Comment:
We always generate both the TO and FROM functions at the same time, so when we apply the protobuf conversion in reverse we have to flip the direction so the function we use switches as well.